File: src/main/java/com/formonk/workers/schedulers/InstaAnalyticsService.java
package com.formonk.workers.schedulers;

import com.ForMonk2.utils.NetworkHandler;
import com.formonk.workers.Utils.Constants;
import org.springframework.stereotype.Service;

import java.io.IOException;
import java.util.HashMap;

@Service
public class InstaAnalyticsService {
    public static void executeService(){
        HashMap params = new HashMap();
        params.put("authToken" , "EAAKZBFNc92isBABhycuaZA4IZA09sIpFYAsl0coTLYv7ZA7iBaqSz8UtmZBsiMbNfLLplgK4ZBZBchZB5oefAiqATMELwjX22CaZAH1IwLXuvZBw12ZAecjOBEvFw4S8PZAZBp9u0aaos2azW24nxkgl7M1AjbUdGQsBY6ZCit9bcfDgrxhd00w46nIk4gubhYnoLZBrk1gobTWgYuKvQZDZD");
        params.put("instaBusinessAccountId" ,"17841409045991210");
        try {
            String response = NetworkHandler.getInstance().sendGet(Constants.FORMONK_ENDPOINT , params);
            System.out.println(response);
        } catch (IOException e) {
            e.printStackTrace();
        }
    }


}

File: src/main/kotlin/com/formonk/workers/schedulers/InstaAnalyticsService.kt
package com.formonk.workers.schedulers

import com.formonk.workers.dao.InstagramAnalyticsLogsDao
import com.formonk.workers.dao.SocialMasterDao
import com.formonk.workers.models.GetUniqueAuthTokens
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.scheduling.annotation.Scheduled
import org.springframework.stereotype.Service

@Service
class InstaAnalyticsService {

    @Autowired
    internal lateinit var  socialMasterDao: SocialMasterDao

    @Autowired
    internal var instagramAnalyticsLogsDao: InstagramAnalyticsLogsDao? = null;

    val tokens: List<GetUniqueAuthTokens>? by lazy{
        socialMasterDao.uniqueAuthTokens
    }

    @Scheduled(fixedDelay = (12 * 60 * 60 * 1000).toLong())
    internal fun executeService(){

        if (tokens != null ){
            print("Not null")
        }else{
            print("Token is null")
        }
    }
}